extensions/import/src/dialogs/derivedColumnDialog.ts

```diff
@@ -0,0 +1,370 @@
+				height: '100%',
+			}).withProps({
+				CSSStyles: {
+					'overflow-y': 'auto',
+					'margin-left': '10px',
+				}
+			}).component();
+
+			this._headerInstructionText = this._view.modelBuilder.text()
+				.withProps({
+					value: constants.headerIntructionText,
+					CSSStyles: {
+						'font-size': 'x-large',
+						'line-height': '22pt',
+						'margin-bottom': '0.7em'
+					}
+				}).component();
+
+			this._bodyInstructionText = this._view.modelBuilder.text()
+				.withProps({
+					value: [
+						constants.deriverColumnInstruction1,
+						constants.deriverColumnInstruction2,
+						constants.deriverColumnInstruction3,
+						constants.deriverColumnInstruction4,
+						constants.deriverColumnInstruction5,
+					],
+					textType: azdata.TextType.OrderedList,
+					CSSStyles: {
+						'font-size': 'large',
+						'line-height': '22pt',
+						'margin-left': '1em',
+						'margin-top': '0em'
+					}
+				}).component();
+
+			this.clearAndAddTransformationContainerComponents(false);
+
+			const flexGrid = view.modelBuilder.flexContainer().withLayout({
+				flexFlow: 'row',
+				height: '100%',
+				width: '100%'
+			}).component();
+
+			/**
+			 * Setting height of the div based on the total viewport height after removing dialog
+			 * header and footer heights. With this the div will occupy the entire page space of the dialog.
+			 */
+			flexGrid.addItem(columnContainer, {
+				flex: '0 0 auto',
+				CSSStyles: {
+					'min-height': 'calc(100vh - 160px)'
+				}
+			});
+			flexGrid.addItem(this._transformationContainer, {
+				flex: '0 0 auto',
+				CSSStyles: {
+					'overflow': 'scroll',
+					'padding-right': '10px',
+					'width': '900px',
+					'max-height': 'calc(100vh - 160px)'
+				}
+			});
+
+			const formBuilder = view.modelBuilder.formContainer().withFormItems(
+				[
+					{
+						component: flexGrid
+					}
+				],
+				{
+					horizontal: false
+				}
+			);
+			const form = formBuilder.withLayout({ width: '100%' }).component();
+			return view.initializeModel(form);
+		});
+
+		this._dialogObject.okButton.onClick(e => {
+			this._doneEmitter.emit('done');
+		});
+
+		this._dialogObject.cancelButton.onClick(e => {
+			this._doneEmitter.emit('close');
+		});
+
+		this._dialogObject.content = [tab];
+		azdata.window.openDialog(this._dialogObject);
+		return new Promise((resolve) => {
+			this._doneEmitter.once('done', async () => {
+				try {
+					await this._provider.sendSaveTransformationRequest({
+						derivedColumnName: this.currentDerivedColumnName
+					});
+					resolve({
+						derivedColumnName: this.currentDerivedColumnName,
+						derivedColumnDataPreview: this.currentTransformation
+					});
+				} catch (e) {
+					console.log(e); // Need to have better error handling for saved transformation. However this seems to be mostly a non-issue.
+				}
+			});
+
+			this._doneEmitter.once('close', async () => {
+				resolve(undefined);
+				azdata.window.closeDialog(this._dialogObject);
```

Comment:
I don't think this is needed, the default handler of the close button will close the dialog